lib/ansible/modules/windows/win_robocopy.ps1

```diff
@@ -81,85 +67,73 @@ if ($flags -eq $null) {
     if ($recurse) {
         $robocopy_opts += "/e"
     }
-}
-Else {
+} else {
     ForEach ($f in $flags.split(" ")) {
         $robocopy_opts += $f
     }
 }
 
 $result.flags = $flags
+$result.cmd = "$robocopy $robocopy_opts"
 
-$robocopy_output = ""
-$rc = 0
-If ($check_mode -eq $true) {
-    $robocopy_output = "Would have copied the contents of $src to $dest"
-    $rc = 0
-}
-Else {
-    Try {
-        &robocopy $robocopy_opts | Tee-Object -Variable robocopy_output | Out-Null
-        $rc = $LASTEXITCODE
-    }
-    Catch {
-        $ErrorMessage = $_.Exception.Message
-        Fail-Json $result "Error synchronizing $src to $dest! Msg: $ErrorMessage"
-    }
+Try {
+    $robocopy_output = &robocopy $robocopy_opts
+    $rc = $LASTEXITCODE
+} Catch {
+    Fail-Json $result "Error synchronizing $src to $dest! Msg: $($_.Exception.Message)"
 }
 
-$result.return_code = $rc
+$result.msg = "Success"
 $result.output = $robocopy_output
+$result.return_code = $rc # Backward compatibility
+$result.rc = $rc
```

Comment:
Won't this cause issues if you don't set `$result.failed = $false` on valid return codes that are not 0?